Fix: CI

Ckk3 opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

Fix CI tests by updating GitHub Actions workflows.
One key point to note: the Python 3.8 and 3.9 tests are consistently failing, and this is not related to the GitHub Actions workflow. I verified that we had previously identified this issue, as referenced in a [comment by Philaeux](https://github.com/strawberry-graphql/strawberry-sqlalchemy/pull/137#issuecomment-2114294219).

The issue was resolved by adding the `name` parameter to the BigInt scalar definition. This prevents an error from being raised when `strawberry.scalar` is called and `__name__` is searched for inside a `Union` type in [this part of the Strawberry code](https://github.com/strawberry-graphql/strawberry/blob/943736fcd651a9aa6994588c92e6e7380a34af39/strawberry/types/scalar.py#L98).

## Types of Changes

<!--- What types of changes does your pull request introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Fix CI tests by updating GitHub Actions workflows to use the latest versions of actions and dependencies, including actions/checkout, actions/setup-python, and codecov/codecov-action. Replace pipx with pip for dependency installation and adjust Python version specifications and caching strategies.

Bug Fixes:
- Fix CI tests by updating GitHub Actions workflows to use the latest versions of actions and dependencies.

CI:
- Update GitHub Actions workflows to use actions/checkout@v4 and actions/setup-python@v5.
- Replace pipx with pip for installing poetry, nox, and nox-poetry.
- Update codecov/codecov-action to version v4.
- Adjust Python version specifications and caching strategies in CI workflows.

<!-- Generated by sourcery-ai[bot]: end summary -->

botberry commented:
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)
<!-- action-check: release-file -->

Ckk3 commented:
Ready for Review!

I’ve updated the CI and incorporated a lot from the Strawberry repository.

One key point to note: the Python 3.8 and 3.9 tests are consistently failing, and this is not related to the GitHub Actions workflow. I verified that we had previously identified this issue, as referenced in a [comment by Philaeux](https://github.com/strawberry-graphql/strawberry-sqlalchemy/pull/137#issuecomment-2114294219).

The issue was resolved by adding the `name` parameter to the BigInt scalar definition. This prevents an error from being raised when `strawberry.scalar` is called and `__name__` is searched for inside a `Union` type in [this part of the Strawberry code](https://github.com/strawberry-graphql/strawberry/blob/943736fcd651a9aa6994588c92e6e7380a34af39/strawberry/types/scalar.py#L98).

patrick91 reviewed (approved):
Nice, thanks!
